include/deepstream/core/event.hpp

```diff
@@ -78,7 +72,7 @@ struct Event {
      * With this constructor instead of `Event(deepstream::Client*)` it
      * becomes easier to test this module.
      */
-    explicit Event(const SendFn&);
+    explicit Event(const SendFn &, SubscriptionId &);
```

Comment:
The other option would be to inject a reference to `Client`, but that would require creating a mock client to unit test `Event`, so... yes?
... but the loose coupling this provides also seems like nice design, just not particularly OO.
If you mean the `SubscriptionId`, that's a reference to a core counter, so that we can ensure ids are globally unique.